src/block_sizes.jl

```diff
@@ -53,11 +53,7 @@ function block_sizes(::Type{T}) where {T}
     st = VectorizationBase.static_sizeof(T)
 
     L2 = (StaticInt{_L2}() - StaticInt{_L1}()) ÷ st
-    if 2_L2 * L2c > _L3 * L3c
-        L3 = StaticInt{_L3}() ÷ st
-    else
-        L3 = (StaticInt{_L3}() - StaticInt{_L2}()) ÷ st
-    end
+    L3 = _calculate_L3(_L2, L2c, _L3, L3c, st)
```

Comment:
Use
```julia
L3 = _calculate_L3(StaticInt{_L2}(), StaticInt{_L3}(), st)
```
here, and then define

```julia
function _calculate_L3(_L2, _L3, st)
    if VectorizationBase.CACHE_INCLUSIVITY[3]
        (_L3 - _L2) ÷ st
    else
        _L2 ÷ st
    end
end
```

Reply:
This no longer relys on constant propagation, since we convert to `StaticInt`s before calling `_calculate_L3`.

Reply:
Alright, I just pushed those changes. Take a look now just as a final double check.

```julia
julia> @code_typed Octavian.block_sizes(Float64)
CodeInfo(
1 ─     nothing::Nothing
│       nothing::Nothing
└──     goto #3 if not false
2 ─     nothing::Nothing
3 ┄     return (Static(72), Static(284), Static(984))
) => Tuple{ArrayInterface.StaticInt{72},ArrayInterface.StaticInt{284},ArrayInterface.StaticInt{984}}
```